code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,188 @@
+/**
+  *Text-to-speech subsystem
+  *
+  *Converts say() speech into a played sound
+  *
+  *Basically works by recieving a file from the say_tts proc,
+  *then feeding it into a TTS generator.
+  *The generator then spits out an .ogg and a .meta files
+  *.ogg is file is then played and .meta is used to measure the length of the speech,
+  *which determines the length of the timer for spam limiting and resets the HUD icon
+  *After the sound is played both files are deleted and the cycle begins anew
+  */
+
+#define GENERATOR_PATH    "tools\\tts_generator\\"	//TTS generator file location
+#define DATA_PATH         GENERATOR_PATH + "data\\"	//Temp files location
+#define STATUS_NEW        0
+#define STATUS_GENERATING 1
+#define STATUS_PLAYING    2
+
+SUBSYSTEM_DEF(tts)
+	name = "Text-to-Speech"
+	wait = 2
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	/// List of items to process
+	var/list/processing
+
+/datum/controller/subsystem/tts/Initialize()
+	LAZYINITLIST(processing)
+
+	if (!CONFIG_GET(flag/enable_tts))
+		can_fire = FALSE
+	else
+		start_engine()
+
+	return ..()
+
+/**
+  *Launches the actual TTS generator
+  */
+/datum/controller/subsystem/tts/proc/start_engine()
+	if (!CONFIG_GET(flag/enable_tts))
+		return
+	var/cmd = "cmd /c start \"tts_generator\" [GENERATOR_PATH]tts_generator.exe"
+	shell(cmd)
+
+/datum/controller/subsystem/tts/proc/check_processing(client/C)
+	if (!C)
+		return FALSE
+
+	for (var/datum/tts/T in processing)

```

Comment:
```suggestion
	for (var/t in processing)
		var/datum/tts/tts_datum = t
```
If you type cast inside a `for` you add a type check.
We want to both avoid the type check and the silent error that comes with it in case the list is polluted.